packages/jsii/lib/compiler.ts

```diff
@@ -42,48 +43,71 @@ export interface CompilerOptions {
     watch: boolean;
 }
 
+export interface TypescriptConfig {
+    compilerOptions: ts.CompilerOptions;
+    include?: string[];
+    exclude?: string[];
+    references?: ts.ProjectReference[];
+}
+
 export class Compiler implements Emitter {
-    public constructor(private readonly options: CompilerOptions) {}
+    private readonly compilerHost: ts.CompilerHost;
+    private typescriptConfig: TypescriptConfig;
+    private rootFiles: string[] = [];
+    private readonly configPath: string;
+
+    public constructor(private readonly options: CompilerOptions) {
+        this.compilerHost = ts.createCompilerHost(COMPILER_OPTIONS);
+        this.compilerHost.getCurrentDirectory = () => this.options.projectInfo.projectRoot;
+        this.configPath = path.join(this.options.projectInfo.projectRoot, 'tsconfig.json');
+    }
 
     /**
      * Compiles the configured program.
      *
      * @param files can be specified to override the standard source code location logic. Useful for example when testing "negatives".
      */
     public async emit(...files: string[]): Promise<EmitResult | never> {
+        await this.buildTypeScriptConfig();
```

Comment:
Yes. They're two different operations.

Reply:
Are they used elsewhere in the core individually?

Reply:
No, but I don't think this is a very good argument.

Do you propose to not make helper functions at all and just inline all code of every project into a 1000-line `main()` function monstrosity, just because it so happens that every function is only called once?

They're separate because determining what the configuration should be, and writing the configuration to disk are two distinct operations.